pythonwhat/checks/has_funcs.py

```diff
@@ -349,15 +349,19 @@ def has_expr(
         if isinstance(eval_sol, str):
             fmt_kwargs["sol_eval"] = '\'{}\''.format(fmt_kwargs["sol_eval"])
 
+    # reformat student evaluation string if it is too long or contains newlines
+    if incorrect_msg == DEFAULT_INCORRECT_MSG:
+        fmt_kwargs["stu_eval"] = utils.shorten_string(fmt_kwargs["stu_eval"])
+
     # check if student or solution evaluations are too long or contain newlines
     if incorrect_msg == DEFAULT_INCORRECT_MSG and (
-        utils.unshowable_string(fmt_kwargs["stu_eval"])
-        or utils.unshowable_string(fmt_kwargs["sol_eval"])
-        or fmt_kwargs["stu_eval"] == fmt_kwargs["sol_eval"]
-    ):
-        fmt_kwargs["stu_eval"] = None
-        fmt_kwargs["sol_eval"] = None
-        incorrect_msg = "Expected something different."
+        len(fmt_kwargs["sol_eval"]) > 50 or
+        utils.has_newline(fmt_kwargs["stu_eval"]) or
+        utils.has_newline(fmt_kwargs["sol_eval"]) or
+        fmt_kwargs["stu_eval"] == fmt_kwargs["sol_eval"]):
+            fmt_kwargs["stu_eval"] = None
```

Comment:
The indentation here seems incorrect (but will be correct if you put this if in the if above).